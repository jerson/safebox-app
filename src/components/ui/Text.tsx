import React from 'react';
import {
  Text as TextBase,
  TextProps as TextBaseProps,
  TextStyle,
  StyleSheet
} from 'react-native';
import Font, {
  FontWeight,
  FontTypeStyle,
  FontFamily
} from '../../modules/resources/Font';

const styles = StyleSheet.create({
  text: {} as TextStyle
});

export interface TextProps extends TextBaseProps {
  weight?: FontWeight;
  textStyle?: FontTypeStyle;
  family?: FontFamily;
  size?: number;
  children: string | React.ReactNode;
}

function Text({ style, weight, textStyle, family, size, ...props }: TextProps) {
  const customStyle = {
    fontSize: size,
    ...Font({
      weight,
      family,
      style: textStyle
    })
  } as TextStyle;
  return <TextBase style={[customStyle, styles.text, style]} {...props} />;
}

export default Text;
