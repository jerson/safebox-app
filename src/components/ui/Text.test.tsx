import React from 'react';
import * as renderer from 'react-test-renderer';
import Text, { TextProps } from './Text';

const props: TextProps = {
  children: 'sample'
};
test('renders without crashing', () => {
  const snapshot = renderer.create(<Text {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<Text {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
