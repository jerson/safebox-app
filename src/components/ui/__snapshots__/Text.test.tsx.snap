// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Nunito",
        "fontSize": undefined,
        "fontStyle": "normal",
        "fontWeight": "400",
      },
      Object {},
      undefined,
    ]
  }
>
  sample
</Text>
`;
